src/main/scala/mesosphere/marathon/core/task/TaskStateChange.scala

```diff
@@ -0,0 +1,27 @@
+package mesosphere.marathon.core.task
+
+import mesosphere.marathon.core.instance.Instance
+
+sealed trait TaskStateChange
+
+object TaskStateChange {
+  case class Update(newState: Task, oldState: Option[Task]) extends TaskStateChange
+  case class Expunge(task: Task) extends TaskStateChange
+  case class NoChange(taskId: Instance.Id) extends TaskStateChange
+  case class Failure(cause: Throwable) extends TaskStateChange
+  object Failure {
+    def apply(message: String): Failure = Failure(TaskStateChangeException(message))
+  }
+}
```

Comment:
Seems like it's a pattern from actor messages, but otherwise I don't really see the point of wrapping the subtypes in anything. The code generator had to do that only because there were name conflicts without it, but in general, I think it's cleaner without any wrapping.